omod/src/main/java/org/openmrs/module/idgen/rest/resource/AutoGenerationOptionResource.java

```diff
@@ -0,0 +1,87 @@
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.FullRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.AlreadyPaged;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.MetadataDelegatingCrudResource;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+@Resource(name = RestConstants.VERSION_1 + IdgenRestController.IDGEN_NAMESPACE + "/autogenerationoption", supportedClass = AutoGenerationOption.class, supportedOpenmrsVersions = {
+        "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*" })
+public class AutoGenerationOptionResource extends MetadataDelegatingCrudResource<AutoGenerationOption> {
+  @Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		DelegatingResourceDescription description = null;
+		
+		// NOTE: description for ref representation is provided by MetaDataDelegatingCrudResource
+		if (rep instanceof DefaultRepresentation || rep instanceof FullRepresentation) {
+      description = new DelegatingResourceDescription();
+      
+			description.addProperty("identifierType");
+			description.addProperty("location");
+			description.addProperty("source");
+			description.addProperty("manualEntryEnabled");
+			description.addProperty("automaticGenerationEnabled");
+		}
+		
+		return description;
+  }
+  
+  @Override
+  public DelegatingResourceDescription getCreatableProperties() {
+      DelegatingResourceDescription description = new DelegatingResourceDescription();
+      description.addProperty("identifierType");
+      description.addProperty("location");
+			description.addProperty("source");
+			description.addProperty("manualEntryEnabled");
+			description.addProperty("automaticGenerationEnabled");
+      return description;
+  }
+
+  @Override
+  public DelegatingResourceDescription getUpdatableProperties() {
+      DelegatingResourceDescription description = new DelegatingResourceDescription();
+      description.addProperty("location");
+			description.addProperty("source");
+			description.addProperty("manualEntryEnabled");
+			description.addProperty("automaticGenerationEnabled");
+      return description;
+  }
+
+  @Override
+  public AutoGenerationOption newDelegate() {
+    return new AutoGenerationOption();
+  }
+
+  @Override
+  public AutoGenerationOption save(AutoGenerationOption autoGenerationOption) {
+    return Context.getService(IdentifierSourceService.class).saveAutoGenerationOption(autoGenerationOption);
+  }
+
+  @Override
```

Comment:
try using this here  `@Override
		                  public AutoGenerationOption getByUniqueId(String uniqueId) {
				  return Context.getService(IdentifierSourceService.class).getAutoGenerationOption(Integer.parseInt(uniqueId));
		    }`

Reply:
Very good suggestion @kwahalf 👍